libs/SalesforceSDK/src/com/salesforce/androidsdk/rest/ClientManager.java

```diff
@@ -417,11 +428,12 @@ public String getNewAuthToken() {
                         if (Looper.myLooper() == null) {
                             Looper.prepare();
                         }
+                        boolean showLoginPage = accounts.length > 1;
                         // Note: As of writing (2024) this call will never succeed because revoke API is an
                         // authenticated endpoint.  However, there is no harm in attempting and the debug logs
                         // produced may help developers better understand the state of their app.
                         SalesforceSDKManager.getInstance()
-                                .logout(null, null, false, OAuth2.LogoutReason.REFRESH_TOKEN_EXPIRED);
+                                .logout(matchingAccount, null, showLoginPage, OAuth2.LogoutReason.REFRESH_TOKEN_EXPIRED);
```

Comment:
`logout` will default to the current user if null is passed 😬.  That is intended in most cases, but really bad in this one.